torchao/quantization/quantize_/workflows/int4/int4_plain_int32_tensor.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import List
+
+import torch
+
+from torchao.quantization.quant_primitives import (
+    MappingType,
+    choose_qparams_affine,
+    quantize_affine,
+)
+from torchao.utils import (
+    TorchAOBaseTensor,
+)
+
+__all__ = [
+    "Int4PlainInt32Tensor",
+]
+
+aten = torch.ops.aten
+
+
+class Int4PlainInt32Tensor(TorchAOBaseTensor):
+    """
+    int4 weight-only quantization on XPU with oneDNN as backend (groupwise quantization only)
+
+    Tensor Attributes:
+        qdata: (N, K/8), packed int4 weight, the data type is int32 here with 4*(int4*2)
+        scale: (K/group_size, N), dtype is the same as the original Tensor dtype
```

Comment:
btw, might also be good to specify what is the supported and optimized original tensor dtype as well, I feel we have been a bit vague on this for all these tensors